src/Microsoft.AspNet.OData.Shared/Query/Validators/ODataQueryValidator.cs

```diff
@@ -72,6 +72,12 @@ public virtual void Validate(ODataQueryOptions options, ODataValidationSettings
                 }
             }
 
+            if (options.SkipToken != null)
+            {
+                ValidateQueryOptionAllowed(AllowedQueryOptions.SkipToken, validationSettings.AllowedQueryOptions);
+                options.SkipToken.Validate(validationSettings);
```

Comment:
looks like we're validating query option twice; can remove redundant check from SkipToken.Validate.